src/System.Net.WebSockets.Client/src/System/Net/WebSockets/WinRTWebSocket.cs

```diff
@@ -515,5 +539,51 @@ private void UpdateStateCloseReceived()
             }
         }
         #endregion
+
+        #region Helpers
+        private static bool InitMessageWebSocketClientCertificateSupported()
+        {
+            return ApiInformation.IsPropertyPresent(
+                "Windows.Networking.Sockets.MessageWebSocketControl",
+                "ClientCertificate");
+        }
+
+        // There are currently only two ways to convert a .NET X509Certificate object into a WinRT Certificate without
```

Comment:
ClientWebSocketOptions.ClientCertificates is of type X509CertificateCollection and all of its methods deal with X509Certificate objects. If it were an X509Certificate2Collection, we'd be dealing with X509Certificate2 objects.